pkgs/test/texlive/default.nix

```
@@ -321,4 +321,46 @@
         echo "tested $binCount binCount: $ignoredCount ignored, $brokenCount broken, $failedCount failed"
         [[ $failedCount = 0 ]]
       '';
+
+  # verify that the precomputed licensing information in default.nix
+  # does indeed match the metadata of the individual packages.
+  #
+  # This is part of the test suite (and not the normal evaluation) to save
+  # time for "normal" evaluations. To be more in line with the other tests, this
+  # also builds a derivation, even though it is essentially an eval-time assertion.
+  licenses =
+    let
+        concatLicenses = builtins.foldl' (acc: el: if builtins.elem el acc then acc else acc ++ [ el ]);
+        savedLicenses = scheme: scheme.meta.license;
+        correctLicenses = scheme: builtins.foldl'
+                (acc: pkg: concatLicenses acc (lib.toList (pkg.meta.license or [])))
+                []
+                scheme.passthru.packages;
+        leq = (a: b: a.fullName <= b.fullName);
+        hasLicenseMismatch = scheme:
+          (lib.isDerivation scheme) &&
+          (lib.sort leq (savedLicenses scheme) != lib.sort leq (correctLicenses scheme));
```

```suggestion
          (lib.sort leq (savedLicenses scheme) != correctLicenses scheme);
```
Let's assert that the list is ordered.